cmd/kube-proxy/app/server.go

```diff
@@ -399,6 +399,11 @@ func NewProxyServer(config *componentconfig.KubeProxyConfiguration, cleanupAndEx
 	proxyMode := getProxyMode(string(config.Mode), iptInterface, iptables.LinuxKernelCompatTester{})
 	if proxyMode == proxyModeIPTables {
 		glog.V(0).Info("Using iptables Proxier.")
+		nodeIP := net.ParseIP(config.BindAddress)
+		if local, _ := isLocalIP(nodeIP); !local {
```

Comment:
Two concerns here, if no bind address is specified. First, it is not obvious that the nil nodeIP will fail on the Equal() test in isLocalIP(). Having an explicit test for nil would be more readable. Second, the corresponding message on L404  would be misleading - as it would say "bind-address  must be local ip".